src/quo2/components/community/token_gating.cljs

```diff
@@ -0,0 +1,140 @@
+             :header-avatar {:margin-right 12}
+             :header-community-avatar-image {:width 32 :height 32}
+             :header-text {:flex 1}
+             :header-info-button {:margin-left 8}
+             :token-requirement-list-spacing {:padding-vertical 8}
+             :token-list {:align-items :center}
+             :token-row {:flex-direction :row}
+             :token-tag-spacing {:margin-left 7}
+             :divider {:height 1
+                       :background-color quo2-colors/neutral-10
+                       :margin-horizontal -20
+                       :margin-top 12
+                       :margin-bottom 8}
+             :enter-button {:margin-top 16}
+             :info-text {:margin-top 12
+                         :color quo2-colors/neutral-50
+                         :text-align :center
+                         :padding-horizontal 20}})
+
+(defn is-token-requirement-met? [token-requirement-list] (reduce
+                                                          #(or (and %1 (:is-sufficient %2)) false)
+                                                          true
+                                                          token-requirement-list))
+
+(defn token-requirement-list [props]
+  (let [{:keys [gate]} props
+        [gate-type token-requirement-list] gate
+        is-sufficient? (is-token-requirement-met? token-requirement-list)
+        message (if
+                 (= gate-type :join)
+                  (str "Join requirements" (when-not is-sufficient? " not") " met")
+                  (if (= gate-type :read)
+                    (str "View only requirements" (when-not is-sufficient? " not") " met")
+                    (str "View and post requirements" (when-not is-sufficient? " not") " met")))]
+    [rn/view
+     [rn/view {:style (get styles :token-requirement-list-spacing)}
+      [text/text {:weight :medium
+                  :number-of-lines 1} message]
+      [text/text {:size :paragraph-2
+                  :number-of-lines 1} "You must hodl:"]]
+
+     [rn/view {:style (merge (get styles :token-row) (merge (get styles :token-requirement-list-spacing)) (get styles :token-list))}
+      (map-indexed (fn [token-index token]
+                     ^{:key token}
+                     [:<>
+                      [rn/view {:style (when-not (= token-index 0) (get styles :token-tag-spacing))}
+                       [token-tag/token-tag {:token (:token token)
+                                             :value (:amount token)
+                                             :size :big
+                                             :is-required (:is-sufficient token)
+                                             :is-purchasable (:is-purchasable token)
+                                             :token-img-src
+                                             (js/require
+                                              (str
+                                               "../resources/images/tokens/mainnet/"
```

Comment:
hmm this seems like it should not be done this way. 

Perhaps you can move this js/require into the screens file so we can use it for the examples and just pass the `img-src` as a prop?

Reply:
Do you think? Otherwise we would have to pass the `token` and `token-img-src` props to the gate's token requirements.

Shouldn't this be as generic as possible?

Reply:
I might be missing context on this, but is it just for tokens that we have in the `resources` folder or is possible that tokens (more so their images) are loaded in from some endpoint?